ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,239 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let create =
+  call ~name:"create" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Create a new row in the Tracing database and register the provider"
+    ~params:
+      [
+        (String, "name_label", "The tracer provider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the tracer provider is to be created on"
+        )
+      ; ( Bool
+        , "enabled"
+        , "Whether the tracer provider is gathering traces or not"
+        )
+      ; ( Map (String, String)
+        , "tags"
+        , "The user tags that identify the tracer provider"
+        )
+      ; ( Set String
+        , "endpoints"
+        , "The file/HTTP endpoints the provider sends traces to"
+        )
+      ; ( Set String
+        , "components"
+        , "Dom0 componenets on which the provider is enabled"
+        )
+      ; ( Set String
+        , "filters"
+        , "Operations that tracers (of this provider) will create spans for"
+        )
+      ; (Set String, "processors", "Pre export operations to traces")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let register =
+  call ~name:"register" ~in_oss_since:None ~lifecycle:[]
```

Comment:
This is done as creation relies on the `uuid` as the key for the new provider, and create should also handle the process of creating the DB object. This will not work in one function with the need to distribute the create call to multiple hosts.

Reply:
OK, so it is an internal function for xapi's use, not meant to be called by external API clients. Then you should define it with `~hide_from_docs:true`.

Reply:
Added this